### ISSUE TITLE
`add_header()` fails if `to` is a factor 

### ISSUE THREAD

**Issue by ccsarapas** [ccsarapas/lighthouse#3]
reprex:
``` r
library(lighthouse)
library(tidyverse)

fakedata <- expand_grid(
  group = c("Group A", "Group B"),
  variable = c("Var X", "Var Y")
)

# both characters - works OK
fakedata %>% 
  add_header(group, variable, indent = "  ")
#> # A tibble: 6 x 1
#>   variable 
#>   <chr>    
#> 1 "Group A"
#> 2 "  Var X"
#> 3 "  Var Y"
#> 4 "Group B"
#> 5 "  Var X"
#> 6 "  Var Y"

# `from` as factor - works OK
fakedata %>% 
  mutate(group = factor(group)) %>% 
  add_header(group, variable, indent = "  ")
#> # A tibble: 6 x 1
#>   variable 
#>   <chr>    
#> 1 "Group A"
#> 2 "  Var X"
#> 3 "  Var Y"
#> 4 "Group B"
#> 5 "  Var X"
#> 6 "  Var Y"

# `to` as factor - fails  
fakedata %>% 
  mutate(variable = factor(variable)) %>% 
  add_header(group, variable, indent = "  ")
#> Error in `dplyr::mutate()`:
#> ! Problem while computing `variable = dplyr::if_else(..SKIP, variable,
#>   stringr::str_c(indent, variable))`.
#> Caused by error in `dplyr::if_else()`:
#> ! `false` must be a `factor` object, not a character vector.
```

<sup>Created on 2022-05-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by ccsarapas** [comment]
Issue persists even if both `from` and `to` are factors.

``` r
fakedata %>% 
  mutate(group = factor(group), variable = factor(variable)) %>% 
  add_header(group, variable, indent = "  ")
#> Error in `dplyr::mutate()`:
#> ! Problem while computing `variable = dplyr::if_else(..SKIP, variable,
#>   stringr::str_c(indent, variable))`.
#> Caused by error in `dplyr::if_else()`:
#> ! `false` must be a `factor` object, not a character vector.
```

<sup>Created on 2022-05-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by ccsarapas** [comment]
This seems to have resolved itself, likely due to v1.1.0 changes to dplyr::if_else() and support for type coercion using vctrs.
``` r
library(lighthouse)
library(tidyverse)

fakedata <- tidyr::expand_grid(
  group = c("Group A", "Group B"),
  variable = c("Var X", "Var Y")
)

# `to` as factor - now works OK
fakedata %>% 
  dplyr::mutate(variable = factor(variable)) %>% 
  add_header(group, variable, indent = "  ")
#> # A tibble: 6 × 1
#>   variable 
#>   <chr>    
#> 1 "Group A"
#> 2 "  Var X"
#> 3 "  Var Y"
#> 4 "Group B"
#> 5 "  Var X"
#> 6 "  Var Y"

# `to` and `from` as factors - now works OK
fakedata %>% 
  dplyr::mutate(group = factor(group), variable = factor(variable)) %>% 
  add_header(group, variable, indent = "  ")
#> # A tibble: 6 × 1
#>   variable 
#>   <chr>    
#> 1 "Group A"
#> 2 "  Var X"
#> 3 "  Var Y"
#> 4 "Group B"
#> 5 "  Var X"
#> 6 "  Var Y"
```

<sup>Created on 2024-07-06 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
